arduino/cores/cores.go

```diff
@@ -247,6 +258,18 @@ func (platform *Platform) GetAllReleasesVersions() []*semver.Version {
 	return versions
 }
 
+// GetAllCompatibleReleasesVersions returns all the version numbers in this Platform Package that contains compatible tools.
+func (platform *Platform) GetAllCompatibleReleasesVersions() []*semver.Version {
+	versions := []*semver.Version{}
+	for _, release := range platform.Releases {
+		if release.Incompatible {
+			continue
+		}
+		versions = append(versions, release.Version)
+	}
+	return versions
+}
```

Comment:
`*Platform.GetAllCompatibleReleasesVersions() []*semver.Version` is used only in
`*Platform.latestCompatibleReleaseVersion() *semver.Version` that is used only in
`*Platform.GetLatestCompatibleRelease() *PlatformRelease`.

At this point, I'd like to drop the first two methods and implement directly the last one (we can cycle directly through the PlatformReleases and avoid a call to `platform.FindReleaseWithVersion(latestVersion)` to retrieve the PlatformRelease from the version.